app/controllers/registration_controller.rb

```diff
@@ -22,13 +22,20 @@ def new
   end
 
   def destroy
-    @registration = Registration.find(params[:registration_id])
-    if current_user == @registration.user && @registration.event.start_date.future?
+    if params[:registration_id].present?
+      @registration = Registration.find(params[:registration_id])
+    else
+      event_id = params[:event_id]
+      user_id = params[:user_id]
+      @registration = Registration.where(user_id: user_id, event_id: event_id).first
```

Comment:
To avoid this conditional statement we can change logic a little bit and pass `user_id or guest_id` as the parameter which along with `id (currently your event_id)` will be used to delete the registration in both the cases.  One thing we should always try to avoid is to keep our controller as slim as possible. Rule of thumb is to keep controller action up to a maximum of 5 lines. Let me know if I wasn't clear enough